lib/hashable.rb

```diff
@@ -1,14 +1,28 @@
 module Hashable
 
-
-
-    def ross_test
-      puts "fly like an eagle"
+    def hash_of_hashes(collection, key1, key2, key3, value2, value3, arg2 = nil )
+      # {key1 => { key2 => value2, key3 => value3}} across collection. arg2 is ignored if not needed
+      # arg2 is optional if the passed method requires arguments
+      hash_of_hashes = Hash.new { |hash, key| hash[key] = {key2 => 0, key3 => 0}}
+      collection.each do |game|
+        hash_of_hashes[game.send(key1)][key2] += game.send(value2) if arg2.nil?
+        hash_of_hashes[game.send(key1)][key2] += game.send(value2, arg2) if !arg2.nil?
+        hash_of_hashes[game.send(key1)][key3] += value3 if value3.is_a?(Numeric)
+        hash_of_hashes[game.send(key1)][key3] += game.send(value3) if !value3.is_a?(Numeric)
+      end
+      hash_of_hashes
     end
```

Comment:
creates a hash by iterationg over `collection`. A real example of away goals per game per season
{20122013 => {:goals => 10, :games_played => 20},
20132014 => {:goals =>15, :games_played => 25}}
```
      hash_of_hashes = Hash.new { |hash, key| hash[key] = {:goals => 0, :games_played => 0}}
      all.each do |game|
        hash_of_hashes[game.season][:goals] += game.away_goals
        hash_of_hashes[game.season)][:games_played] += 1
      end
      hash_of_hashes
```